packages/svelte/src/internal/client/reactivity/effects.js

```diff
@@ -325,7 +325,7 @@ export function template_effect(fn) {
 			value: '{expression}'
 		});
 	}
-	return render_effect(fn);
+	return block(fn);
```

Comment:
I'm confused, can we just change the type of effect? I would've expected this has unintended side effects.

Reply:
There's pretty much the same thing

Reply:
In what ways are they not? If `template_effect` which is pretty much used for all things that are output into the html isn't using a render effect, then what do we even need it for?

Also did you see my comment about only doing the validation in runes mode?

Reply:
The validation was always runes mode anyway, we're re-using the same validation as before. Blocks differ in two ways:

- They don't clear destroy branch effects on updates
- They disallow mutations as do deriveds

Reply:
Asked differently: what's stopping us from keep using the render effect, if we would extract the validation into a common method?

Reply:
I'm curious what you mean by validation here? There is no actual validation. We just block calling `set` on a signal during a reaction that is derived or a block effect. This is how it works in other places.

Reply:
Sorry, didn't see how it came to be in the other PR. Can we block the set when it's a render effect? Wouldn't that be more in line what the related issue is about anyway? Or does that have unintended consequences?

Reply:
Unfortunately `$effect.pre` and a bunch of other stuff already using render effect with mutations.

Reply:
Can this change affect or be affected in some by way the work done in https://github.com/sveltejs/svelte/pull/13600 ?

Reply:
I don't think they're connected.

Reply:
I'm with @dummdidumm on this — treating every template expression as a block effect seems like it could have undesirable consequences, if not now then in the future. For example it means that in `update_effect` we do `destroy_block_effect_children` instead of `destroy_effect_children`, which means we don't destroy child deriveds, which feels like it could be bad.

Shouldn't we just have a new flag that means 'state mutations not allowed' and apply that to deriveds and blocks and template expressions?

Then in `set` we'd do something like this

```diff
-(active_reaction.f & (DERIVED | BLOCK_EFFECT)) !== 0 &&
+(active_reaction.f & (MUTATIONS_DISALLOWED)) !== 0 &&
```

Reply:
> which means we don't destroy child deriveds, which feels like it could be bad.

That's an oversight, we should always be destroying them. I can put up a PR for that tomorrow.